File: ios/Harakat Lens/Services/Quran/QuranAyah.swift
//
//  QuranAyah.swift
//  Harakat Lens

import Foundation

struct QuranAyah: Decodable, Identifiable, Hashable {
    let surah: Int
    let ayah: Int
    let text: String
    let normalized: String
    let transliteration: String
    let translationEn: String

    var id: String {
        "\(surah):\(ayah)"
    }

    enum CodingKeys: String, CodingKey {
        case surah
        case ayah
        case text
        case normalized
        case transliteration
        case translationEn = "translation_en"
    }
}

struct SurahName: Decodable, Identifiable, Hashable {
    let number: Int
    let english: String
    let transliteration: String
    let arabic: String

    var id: Int {
        number
    }
}

File: android/app/src/main/java/dev/veeso/harakatlens/arabic/quran/QuranAyah.kt
package dev.veeso.harakatlens.arabic.quran

/**
 * Ported verbatim from the Harakat-Lens iOS app and the already-complete iOS
 * `ArabicExample/Quran/QuranAyah`.
 */
data class QuranAyah(
    val surah: Int,
    val ayah: Int,
    val text: String,
    val normalized: String,
    val transliteration: String,
    val translationEn: String,
) {
    val id: String get() = "$surah:$ayah"
}
